internal/controllers/app_controller.go

```diff
@@ -270,12 +290,245 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App, logger
+			var dep appsv1.Deployment
+			if err := r.Get(ctx, client.ObjectKey{
+				Namespace: framework.Spec.NamespaceName,
+				Name:      fmt.Sprintf("%s-%s-%d", app.GetName(), process.Name, latestDeployment.Version),
+			}, &dep); err != nil {
+				return reconcileResult{
+					status:  v1.ConditionFalse,
+					message: fmt.Sprintf("failed to get deployment: %v", err),
+				}
+			}
+			err = r.watchDeployEvents(ctx, app, framework.Spec.NamespaceName, &dep, &process, r.Recorder)
+			if err != nil {
+				return reconcileResult{
+					status:  v1.ConditionFalse,
+					message: fmt.Sprintf("failed to get deploy events: %v", err),
+				}
+			}
+		}
+	}
+
 	return reconcileResult{
 		framework: ref,
 		status:    v1.ConditionTrue,
 	}
 }
 
+// watchDeployEvents watches a namespace for events and, after a deployment has started updating, records events
+// with updated deployment status and/or healthcheck and timeout failures
+func (r *AppReconciler) watchDeployEvents(ctx context.Context, app *ketchv1.App, namespace string, dep *appsv1.Deployment, process *ketchv1.ProcessSpec, recorder record.EventRecorder) error {
+	config, err := GetRESTConfig()
+	if err != nil {
+		return err
+	}
+	cli, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	opts := metav1.ListOptions{
+		FieldSelector: "involvedObject.kind=Pod",
+		Watch:         true,
+	}
+	watcher, err := cli.CoreV1().Events(namespace).Watch(ctx, opts) // requires "watch" permission on events in clusterrole
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	// wait for Deployment Generation
+	timeout := time.After(DefaultPodRunningTimeout)
+	for dep.Status.ObservedGeneration < dep.Generation {
+		dep, err = cli.AppsV1().Deployments(namespace).Get(ctx, dep.Name, metav1.GetOptions{})
+		if err != nil {
+			recorder.Eventf(app, v1.EventTypeWarning, appReconcileError, "error getting deployments: %s", err.Error())
+			return err
+		}
+		select {
+		case <-time.After(100 * time.Millisecond):
+		case <-timeout:
+			recorder.Event(app, v1.EventTypeWarning, appReconcileError, "timeout waiting for deployment generation to update")
+			return errors.Errorf("timeout waiting for deployment generation to update")
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+
+	return r.watchFunc(ctx, app, namespace, dep, process, recorder, watcher, cli, timeout)
+}
+
+func (r *AppReconciler) watchFunc(ctx context.Context, app *ketchv1.App, namespace string, dep *appsv1.Deployment, process *ketchv1.ProcessSpec, recorder record.EventRecorder, watcher watch.Interface, cli kubernetes.Interface, timeout <-chan time.Time) error {
+	var err error
+	watchCh := watcher.ResultChan()
+	recorder.Eventf(app, v1.EventTypeNormal, appReconcileStarted, "Updating units [%s]", process.Name)
+
+	var specReplicas int32
+	if dep.Spec.Replicas != nil {
+		specReplicas = *dep.Spec.Replicas
+	}
+	oldUpdatedReplicas := int32(-1)
+	oldReadyUnits := int32(-1)
+	oldPendingTermination := int32(-1)
```

Comment:
this is always -1? 

should we read old status instead?

Reply:
It gets set to `pendingTermination` each cycle here: https://github.com/theketchio/ketch/pull/177/files#diff-01570fd749623ed07d5f2e0b7097495a4ef86b6f3419378180bc43fc73c9223eR443. I'm okay with changing it, but was trying to keep output similar to Shipa's: https://github.com/shipa-corp/shipa/blob/master/provision/kubernetes/app_manager.go#L445